src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -440,7 +440,7 @@ private void RenderEndTagHelpersScope()
                                                   _tagHelperContext.ScopeManagerEndMethodName);
         }
 
-        private void RenderTagOutput(string tagOutputMethodName)
+        private void RenderTagOutput(string tagOutputMethodName, bool isString = false)
```

Comment:
nits / suggestions:
- method is called very only a few times.  don't use default value and make calls explicit.
- parameter sets expectations for what the named (`tagOutputMethodName`) method returns, so perhaps `methodReturnsString`
- remove the design time check and do that for the entire second half of `RenderTagHelper()`